src/io/exporters/FBXGeometry.js

```diff
@@ -0,0 +1,177 @@
+    this.normals = new Float32Array(this.vertsCount * this.itemSize.normal);
+    this.colors = new Float32Array(this.vertsCount * this.itemSize.color);
+    this._extendVertices(geo, info);
+    // number of indices stays the same
+    this.indices = new Uint32Array(index.count);
+    this._extendIndices(geo, info);
+  }
+
+  /** Extend vertex attributes to have one more slice to make sharp middle startColor-endColor line. */
+  _extendVertices(geo, info) {
+    const { position } = geo.attributes;
+    const { normal } = geo.attributes;
+    const geoParams = geo.getGeoParams();
+    const cutRaw = 1; // we expect cylinders of 2 segments in height => so half segment = 1
+    this._cutRawStart = cutRaw * geoParams.radialSegments;
+    this._cutRawEnd = this._cutRawStart + info.addPerCylinder;
+    { // write first half of cylinder
+      let temp = position.array.slice(0, this._cutRawEnd * position.itemSize);
+      this.positions.set(temp, 0);
+      temp = normal.array.slice(0, this._cutRawEnd * normal.itemSize);
+      this.normals.set(temp, 0);
+    }
+    { // write second part of cylinder
+      let temp = position.array.slice(this._cutRawStart * position.itemSize, position.array.length);
+      this.positions.set(temp, this._cutRawEnd * position.itemSize);
+      temp = normal.array.slice(this._cutRawStart * normal.itemSize, normal.array.length);
+      this.normals.set(temp, this._cutRawEnd * normal.itemSize);
+    }
+  }
+
+  /** Shift values of second part (+caps) indices by newly added vertices count. Number of faces remains the same. */
+  _extendIndices(geo, info) {
+    const { index } = geo;
+    const indicesPerQuad = 6; // quad = 2 triangles => 6 indices
+    const startToShift = info.addPerCylinder * indicesPerQuad;
+    const shift = info.addPerCylinder;
+    let shifted = index.array.slice(startToShift, index.count);
+    shifted = shifted.map((x) => x + shift); // shift only the endings
+    this.indices.set(index.array, 0);
+    this.indices.set(shifted, startToShift);
+  }
+
+  /**
+   * Set defined colors: (first part + bottom cap), (second part + top cap)
+   * @param {Object} color1 - THREE.Color.
+   * @param {Object} color2 - THREE.Color.
+   */
+  setColors(color1, color2) {
+    const colorSize = this.itemSize.color;
+    const part1End = this._cutRawEnd * colorSize;
+    const part2End = part1End * 2;
+    const capSize = (this._facesPerSlice + 1) * colorSize;
+    let color = null;
+    let offset = 0;
+    for (let i = 0, l = this.colors.length; i < l; i += colorSize) {
+      if (i < part1End) { // first part
+        color = color1;
+      } else if (i < part2End + capSize) { // second part + top cap
+        color = color2;
+      } else { // bottom cap
+        color = color1;
+      }
+      this.colors[offset++] = color.r;
+      this.colors[offset++] = color.g;
+      this.colors[offset++] = color.b;
+    }
+  }
+}
+
+export default {
+  OneColorGeo,
+  TwoColoredCylinder,
+};
```

Comment:
I think we should not use default export for exporting multiple names unrelated to the name of the module. It is great when we put a single class (or a single function) in a separate file but it is not the case.

BTW, wouldn't you like to put each class in a separate file under `fbx/` subdirectory?

Reply:
Done.